src/tests/QuantumCSSplitting.test.ts

```diff
@@ -229,7 +229,7 @@ export class QuantumCSSSplittingTest {
 				env.fileShouldExist("styles.css.map");
 				env.fileShouldExist("chunks/dcbf7349-ea9fe601.css");
 				env.fileShouldExist("chunks/ea9fe601.css.map");
-				env.fileShouldExist("chunks/c3eff8a7-ea9fe601.js");
```

Comment:
what happened here? ;-)

Reply:
Hash seems to be different. I changed it back, could you point me in the right direction as where to look?

Reply:
But the tests are failing. Are you running it on windows?

Reply:
Hmm, I realized why. If you are in `master` branch and run `QuantumCSSplitting.test.ts` but before in `File.js` on `transpileUsingTypescript` change it to:
```javascript
				const res = ts.transpileModule(this.contents, this.getTranspilationConfig());
				console.dir(res, { compact: false })
				console.dir(this.getTranspilationConfig())
				return res
```
You'll see this:
```javascript
{
  outputText: '"use strict";\nexports.__esModule = true;\n' +
    'require("./foobar.scss");\n//# ' +
    'sourceMappingURL=foobar.js.map',
  diagnostics: [
    {
      file: undefined,
      start: undefined,
      length: undefined,
      messageText: "Argument for '--target' option must " +
        "be: 'es3', 'es5', 'es6', 'es2015', " +
        "'es2016', 'es2017', 'es2018', " +
        "'esnext'.",
      category: 1,
      code: 6046,
      reportsUnnecessary: undefined
    }
  ],
  sourceMapText: '{"version":3,"file":"foobar.js","sourceRoot":"","sources":["foobar.ts"],"names":[],"mappings":";;AACE,yBAAsB","sourcesContent":["\\n\\t\\timport ' +
    '\\"./foobar.scss\\"\\n"]}'
}
{ compilerOptions:
   { module: 'commonjs',
     target: 'ES7',
     sourceMap: true,
     inlineSources: true,
     jsx: 'react',
     importHelpers: true,
     emitDecoratorMetadata: true,
     experimentalDecorators: true },
  fileName:
   '/Users/huangjx/Desktop/eddyw/Github/fuse-box/.fusebox/virtual-files/1548244391810/foobar.ts',
  transformers: {} }
```
That's because on default, we're using `ES7` .. but it's an invalid target. My branch fixes the issue.
`outputText` in `master` for last test:
```javascript
  outputText: '"use strict";\nexports.__esModule = true;\n' +
    'require("./foobar.scss");\n//# ' +
    'sourceMappingURL=foobar.js.map',
```
`outputText` in my branch:
```  outputText: '"use strict";\nObject.defineProperty(exports, ' +
    '"__esModule", { value: true });\n' +
    'require("./foobar.scss");\n//# ' +
    'sourceMappingURL=foobar.js.map',
```

Reply:
I reverted last commit to which changes the hash on Quantum test, we were not transpiling to the correct target.
FuseBox default `compilerOptions` uses `{ target: "ES7" }` which should have been `{ target: "es2016" }`. This branch uses `ts.ScriptTarget` and parses and validates tsConfig, so compilerOptions passed are `{ target: 3 }` which corresponds to `es2016`|`es7`
[File.ts#L18](https://github.com/fuse-box/fuse-box/blob/master/src/core/File.ts#L18)

Since errors were not checked before in CompilerOptions, those were not caught. And since `transpileModule` always transpiles regardless of errors, the output now it wrong:
```javascript
exports.__esModule = true;
require("./foobar.scss");
```
Expected:
```javascript
Object.defineProperty(exports, "__esModule", { value: true });
require("./foobar.scss");```

Reply:
Local simple test:
```const ts = require('ts');
const a = ts.transpileModule('export default 123', { compilerOptions: { module: 'commonjs', target: 'es2016' } }) // <<< expected
const b = ts.transpileModule('export default 123', { compilerOptions: { module: 'commonjs', target: 'ES7' } }) // <<< fails```

Reply:
@nchanged ping 🙈

Reply:
@eddyw sorry ;-) thanks for pinging. Will check it out today

Reply:
@nchanged ping 🙈
Anyway to help?

Reply:
There is some very annoying timing issue (racing condition) with this test which bugs me from time to time (